Source/Core/Core/IOS/FS/FileSystemProxy.h

```diff
@@ -24,12 +24,50 @@ constexpr FS::Fd INVALID_FD = 0xffffffff;
 class FSDevice : public Device
 {
 public:
+  class ScopedFd
+  {
+  public:
+    ScopedFd(FSDevice* fs, s64 fd, Ticks tick_tracker = {})
+        : m_fs{fs}, m_fd{fd}, m_tick_tracker{tick_tracker}
+    {
+    }
+
+    ~ScopedFd()
+    {
+      if (m_fd >= 0)
+        m_fs->Close(m_fd, m_tick_tracker);
+    }
+
+    ScopedFd(ScopedFd&& other) noexcept { *this = std::move(other); }
```

Comment:
Can you implement or delete the copy constructor?

Reply:
Both the copy constructor and copy assignment operators are automatically deleted because there is a user-defined move constructor or move assignment operator.

Reply:
True.  I generally feel like it's good practice to specify them all for completeness, see: https://en.cppreference.com/w/cpp/language/rule_of_three (rule of five)